src/AutoRest.CSharp.V3/Output/Models/Shared/Parameter.cs

```diff
@@ -8,19 +8,22 @@ namespace AutoRest.CSharp.V3.Output.Models.Shared
 {
     internal class Parameter
     {
-        public Parameter(string name, string? description, CSharpType type, Constant? defaultValue, bool isRequired)
+        public Parameter(string name, string? description, CSharpType type, Constant? defaultValue, bool isRequired, bool isArtificial = false)
         {
             Name = name;
             Description = description;
             Type = type;
             DefaultValue = defaultValue;
             IsRequired = isRequired;
+            IsArtificial = isArtificial;
         }
 
         public CSharpType Type { get; }
         public string Name { get; }
         public string? Description { get; }
         public Constant? DefaultValue { get; }
         public bool IsRequired { get; }
+        // Indicates the parameter was created by the generator, not defined in the swagger
+        public bool IsArtificial { get; }
```

Comment:
Required to differentiate a nextLink parameter we create (artificial) versus a nextLink parameter that comes from swagger (natural).

Reply:
What about `IsRawNextLink`?

Reply:
Since this is in `Parameter`, I didn't want it to be to the specific context of `nextLink`/paging. So, it could be used for any non-natural parameters. We have other ones, like `cancellationToken`, we could create in this way, which would help for parameter writing/processing consistency. For nextLink, I only care if it came from swagger or not.

Reply:
Oh, I though this is on the PathSegment, let's move it to be on the path segment.

Reply:
Fixed. PathSegment handles both host and path segments. Because of that, it now has `IsRaw`.